Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -279,33 +293,25 @@ namespace AzToolsFramework
                     }
                 });
 
-                // Create a link between the templates of the newly created instance and the instance it's being parented under.
-                CreateLink(
-                    instanceToCreate->get(), commonRootEntityOwningInstance->get().GetTemplateId(), undoBatch.GetUndoBatch(),
-                    AZStd::move(patch));
+                // Update the prefab template for the new instance.
+                // This update includes new child information and excludes transform data seen from its parent.
+                UpdateTemplateForInstance(newInstanceTemplateId, instanceToCreate->get());
```

Comment:
Discussed about this offline. This can just be limited to updating the newContainerEntityId with latest information rather than updating the entire prefab instance.

Reply:
good callout. The current approach seems like an overkill and it can be reduced to only update what it needs. I'll try out to see if it works.